engine/flink/api/src/main/scala/pl/touk/nussknacker/engine/flink/api/process/FlinkSource.scala

```diff
@@ -1,17 +1,28 @@
 package pl.touk.nussknacker.engine.flink.api.process
 
 import org.apache.flink.api.common.typeinfo.TypeInformation
-import org.apache.flink.streaming.api.functions.TimestampAssigner
+import org.apache.flink.streaming.api.TimeCharacteristic
+import org.apache.flink.streaming.api.functions.{AssignerWithPeriodicWatermarks, AssignerWithPunctuatedWatermarks, TimestampAssigner}
 import org.apache.flink.streaming.api.functions.source.SourceFunction
-import pl.touk.nussknacker.engine.api.MethodToInvoke
+import org.apache.flink.streaming.api.scala.{DataStream, StreamExecutionEnvironment}
+import pl.touk.nussknacker.engine.api.{MetaData, MethodToInvoke}
 import pl.touk.nussknacker.engine.api.process.{Source, SourceFactory}
-import pl.touk.nussknacker.engine.api.test.TestDataParser
 
 /**
   * Source with methods specific for Flink
   * @tparam T - type of event that is generated by this source. This is needed to handle e.g. syntax suggestions in UI
   */
-trait FlinkSource[T] extends Source[T] {
+trait BaseFlinkSource[T] extends Source[T] {
```

Comment:
Please swap names: FlinkSource <> BaseFlinkSource and add the same abstraction layers for tests

Reply:
:+1:  I swapped names, tests are a bit clumsy for now but I don't have really elegant idea for now..

Reply:
Looks good. If you already have changed this api, can you also rename `toFlinkSource`->`flinkSource` to clarify that objects of this class are not converted to flink source but flink source is used for preparing DataStream?